crates/cast/bin/cmd/create2.rs

```diff
@@ -148,9 +158,14 @@ impl Create2Args {
             handles.push(std::thread::spawn(move || {
                 // Read the first bytes of the salt as a usize to be able to increment it.
                 struct B256Aligned(B256, [usize; 0]);
-                let mut salt = B256Aligned(B256::ZERO, []);
+                let mut salt = B256Aligned(top_bytes, []);
                 // SAFETY: B256 is aligned to `usize`.
-                let salt_word = unsafe { &mut *salt.0.as_mut_ptr().cast::<usize>() };
+                let salt_word = unsafe {
+                    let ptr: *mut u8 = &mut salt.0[0];
+                    // Offset to preserve caller address at the beginning of the salt.
+                    // Must offset by 24 and not 20 to align u8s and usize.
+                    &mut *ptr.add(24).cast::<usize>()
```

Comment:
This should be `salt.0.as_mut_ptr().add(32 - usize::BITS * 8)` to account for target ptr size

Reply:
Yup, this works. I thought that the pointer to `FixedBytes<32>` would have a bigger size, so I couldn't do exact pointer arithmetics with `.add`. That's why I got the pointer to the first `u8` in the array, but it's still 1 byte, so we can use it.

Reply:
`as_mut_ptr` is from the `[u8]` deref, which returns a u8 ptr to the first element